### PR TITLE
feat: actual premium consumption statistics from quota snapshots

### PR DESCRIPTION
**Pull request by nick3**
## Summary

- Replace model-multiplier-based `cost_units_sum` with actual Premium quota consumption derived from `quota_snapshots` table
- Stats API now merges snapshot consumption data (via SQLite LAG window function) with request metrics from `daily_premium_stats` / `request_log`
- Add `localtime` timezone handling to snapshot date grouping for consistency with existing hourly stats
- Extend `cleanupStatsRetention()` to also clean `quota_snapshots` table

## Changes

| File | Change |
|---|---|
| `src/lib/stats-store.ts` | Interface rename `cost_units_sum` → `premium_consumed`; rewrite `getDailyPremiumStats`/`getHourlyPremiumStats` to merge snapshot consumption; add `localtime` to `getConsumptionFromSnapshots`; add snapshot retention cleanup; add `localDateToMs`/`mergeMetricsAndConsumption` helpers |
| `tests/stats-store.test.ts` | Fix timezone-sensitive timestamps (bun test UTC vs system UTC+8); update assertions for `premium_consumed` |
| `admin-ui/src/lib/admin-api.ts` | `DailyStatsItem.cost_units_sum` → `premium_consumed` |
| `admin-ui/.../premium-usage-chart.tsx` | Update `dataKey` and tooltip reference |
| `admin-ui/.../account-usage-chart.tsx` | Update pivot field reference |

## Test plan

- [x] 399 tests pass (0 fail)
- [x] TypeScript typecheck clean
- [x] ESLint lint clean
- [x] Full build successful
- [x] Timezone edge case tested: bun test (UTC) + SQLite localtime (system TZ) consistency verified

## 发布说明

* **新功能**
  * 添加配额快照跟踪，记录并存储账户配额快照以便统计。

* **改进**
  * 统计与图表现在显示并使用“高级用量（premium_consumed）”指标，更多基于快照差值计算消耗，提升使用量准确性。
  * 报表按日/小时合并并对缺失数据进行零填充。

* **测试**
  * 更新并扩展迁移与消费计算相关测试，覆盖快照写入与回填场景。

* **Chores**
  * 升级数据库迁移到新版本并支持快照回填。

### CODE REVIEW

File: src/lib/stats-store.ts [hunk 1]
```diff
@@ -64,6 +70,100 @@ export class StatsStore {
...
+            CASE WHEN prev_remaining IS NOT NULL AND prev_remaining > remaining
+                 THEN prev_remaining - remaining
+                 ELSE 0
+            END
+          ) AS premium_consumed
+        FROM with_prev
+        WHERE snapshot_at_ms >= ?
+        GROUP BY 1, 2
+        ORDER BY 1, 2`,
+      )
```
[the remark this answers is not in the document]
Comment (nick3) [reply src/lib/stats-store.ts:164]: Thanks for the thorough review! You're right about the non-determinism concern with duplicate timestamps.

I've added `id` as a tiebreaker to the `LAG()` window's `ORDER BY` clause (`ORDER BY snapshot_at_ms, id`), and propagated the `id` column through the baseline and all_snaps CTEs.

I skipped the `ROW_NUMBER()` rewrite for the baseline CTE since `snapshot_at_ms` has millisecond precision and snapshots are inserted per-account on quota refresh — duplicate timestamps per account are practically impossible. The current `MAX(snapshot_at_ms)` + JOIN approach remains readable and correct for this use case.

Also adopted the other two nitpick suggestions (caching `Date.now()` and adding inline SQL parameter comments) in the same commit: 80315df

File: src/lib/stats-store.ts [hunk 2]
```diff
@@ -157,8 +301,91 @@ export class StatsStore {
...
+      ...consumption.map((c) => `${c.date}|${c.account_id}`),
+    ])
+    const byAccountMetricsMap = new Map(
+      byAccountMetrics.map((m) => [`${m.date}|${m.account_id}`, m]),
+    )
+
+    const byAccount: Array<DailyAccountStats> = [...allAccountDateKeys]
+      .sort()
+      .map((key) => {
+        const [date, account_id] = key.split("|")
```
[the remark this answers is not in the document]
Comment (nick3) [reply src/lib/stats-store.ts:403]: Thanks for flagging this! In this project, account IDs are validated by `validateAccountId()` in `accounts-registry.ts:27` which enforces `/^[a-z0-9-]+$/i` — pipe characters cannot appear in account IDs.

The composite key is constructed and parsed within the same `private` method (`mergeMetricsAndConsumption`), so the delimiter is safe and the data is fully controlled internally.

That said, I appreciate the defensive coding mindset — if account ID formats ever change in the future, we'll revisit this.

[the remark this answers is not in the document]
Comment (nick3) [reply src/lib/stats-store.ts:403]: Same reasoning as above — account IDs are validated by `validateAccountId()` (`accounts-registry.ts:27`) which only allows `[a-z0-9-]`, so pipe characters are impossible in practice.

The key construction and parsing are co-located in the same private method, keeping the delimiter contract internal and safe. Thanks for the suggestion though!

File: src/lib/stats-store.ts [hunk 3]
```diff
@@ -73,30 +178,54 @@ export class StatsStore {
...
+      date: string
+      account_id: string
+      request_count: number
+      tokens_total: number
+      error_count: number
+    }>
 
-    return { daily, byAccount }
+    const fromMs = this.localDateToMs(params.from)
+    const toMs = this.localDateToMs(params.to) + 86_399_999
```
[the remark this answers is not in the document]
Comment (nick3) [reply src/lib/stats-store.ts]: Good catch! While the project primarily runs in UTC+8 (no DST), this is a valid concern for portability.

Fixed by adding a `localDateEndMs()` helper that computes the next local midnight minus 1ms, which correctly handles DST transitions via JavaScript's Date constructor:

```typescript
private localDateEndMs(dateStr: string): number {
  const [y, m, d] = dateStr.split("-").map(Number)
  return new Date(y, m - 1, d + 1).getTime() - 1
}
```

Committed in 073f379.

File: src/lib/stats-store.ts [hunk 4]
```diff
@@ -142,23 +274,122 @@ export class StatsStore {
...
+      const nowMs = Date.now()
+      const cutoffMs = nowMs - retentionDays * 24 * 60 * 60 * 1000
+      const cutoffDate = toLocalDateString(cutoffMs)
       this.db
         .query("DELETE FROM daily_premium_stats WHERE date < ?;")
         .run(cutoffDate)
+
+      this.db
+        .query("DELETE FROM quota_snapshots WHERE snapshot_at_ms < ?;")
+        .run(cutoffMs)
```
[the remark this answers is not in the document]
Comment (nick3) [reply src/lib/stats-store.ts]: Excellent find! You're right — the baseline CTE in `getConsumptionFromSnapshots()` relies on having a snapshot before `fromMs`, and blind cleanup would break that.

Fixed the cleanup to preserve the latest snapshot per account before the cutoff:

```sql
DELETE FROM quota_snapshots
WHERE snapshot_at_ms < ?
  AND id NOT IN (
    SELECT qs.id FROM quota_snapshots qs
    INNER JOIN (
      SELECT account_id, MAX(snapshot_at_ms) AS max_ts
      FROM quota_snapshots WHERE snapshot_at_ms < ?
      GROUP BY account_id
    ) latest ON qs.account_id = latest.account_id
            AND qs.snapshot_at_ms = latest.max_ts
  )
```

Also added a test verifying the baseline preservation behavior. Committed in 073f379.
